src/realm/object-store/object_schema.hpp

```diff
@@ -37,18 +35,16 @@ struct Property;
 
 class ObjectSchema {
 public:
-    using IsEmbedded = util::TaggedBool<class IsEmbeddedTag>;
-    using IsAsymmetric = util::TaggedBool<class IsAsymmetricTag>;
+    /// The type of tables supported by a realm.
+    /// Note: Enumeration value assignments must be kept in sync with <realm/table.hpp>.
+    enum class TableType : uint8_t { TopLevel = 0, Embedded = 0x1, TopLevelAsymmetric = 0x2 };
```

Comment:
should we add a static_assert in the implementation file that verifies the values match? Is the reason we don't want to just use the actual `Table::Type` enum here that we don't want to pull in table.hpp as an include in this file?

also, from a naming perspective. should this really be ObjectType rather than TableType at the ObjectStore layer?

Reply:
I don't have a strong opinion on the name (as a fact I wasn't happy with it), but I think ObjectType makes more sense.

The reason is that since some SDKs interface with OS directly, I find it a bit weird that they would need to know about table.hpp and Tale::Type when creating schemas (I want everything doing with schemas to be contained in object_schema.hpp).